tools/distpartitioning/convert_partition.py

```diff
@@ -117,8 +123,19 @@ def create_dgl_object(graph_name, num_parts, \
     node_map_val = {ntype: [] for ntype in ntypes}
     edge_map_val = {etype.split(":")[1]: [] for etype in etypes}
 
-    shuffle_global_nids, ntype_ids, global_type_nid = node_data[constants.SHUFFLE_GLOBAL_NID], \
-            node_data[constants.NTYPE_ID], node_data[constants.GLOBAL_TYPE_NID]
+    memory_snapshot("CreateDGLObj_AssignNodeData", part_id)
+    shuffle_global_nids = node_data[constants.SHUFFLE_GLOBAL_NID]
+    node_data.pop(constants.SHUFFLE_GLOBAL_NID)
+    gc.collect()
+
+    ntype_ids = node_data[constants.NTYPE_ID]
+    node_data.pop(constants.NTYPE_ID)
+    gc.collect()
+
+    global_type_nid = node_data[constants.GLOBAL_TYPE_NID]
+    node_data.pop(constants.GLOBAL_TYPE_NID)
+    node_data = None
+    gc.collect()
```

Comment:
you only need to call gc once here. you pop data from the map but the data is still referenced by the variable. then when you call gc, it cannot collect memory. only when you dereference `node_data`, you should trigger GC.